File: zeebe/zeebe-cancel-wf-instance/src/main/java/com/webpals/zeebe/cancel/wf/instance/AppConfig.java
/*
 * Copyright (c) 2019. Webpals
 */

package com.webpals.zeebe.cancel.wf.instance;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;

import java.io.File;

/**
 * @author Yuriy Stul
 */
public class AppConfig {
    private static final AppConfig instance = new AppConfig();
    private static final Config config;

    static {
        config = ConfigFactory
                .parseFile(new File("application.conf"))
                .withFallback(ConfigFactory.load());
    }

    private AppConfig(){}

    static AppConfig getInstance() {
        return instance;
    }

    String zeebeHost() {
        return config.getConfig("zeebe").getString("host");
    }

    int zeebePort() {
        return config.getConfig("zeebe").getInt("port");
    }
}

File: netflix-conductor/conductor-worker/src/main/scala/com/stulsoft/poc/conductor/worker/AppConfig.scala
/*
 * Copyright (c) 2019. Yuriy Stul
 */

package com.stulsoft.poc.conductor.worker

import java.io.File

import com.typesafe.config.ConfigFactory

/**
 * @author Yuriy Stul
 */
object AppConfig {
  private lazy val config = ConfigFactory.parseFile(new File("application.conf"))
    .withFallback(ConfigFactory.load())

  def rootUri(): String = {
    s"""http://${config.getConfig("conductor").getString("host")}:${config.getConfig("conductor").getInt("port")}/api/"""
  }

  def threadCounter(): Int = {
    config.getConfig("worker").getInt("threadCounter")
  }
}
